### PR TITLE
Add classes for I/O

### PR DESCRIPTION
**Pull request by maxwalls**

### CODE REVIEW

File: src/main/java/org/mal_lang/langspec/LangReader.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+      }
+      throw (IOException) cause;
+    }
+  }
+
+  private void readIcon(ZipInputStream zipIn, String name) throws IOException {
+    // Remove "lang/" prefix
+    var fileName = name.substring(6);
+    if (!fileName.endsWith(".png") && !fileName.endsWith(".svg")) {
+      return;
```
Comment (mikaelforeseeti) [inline src/main/java/org/mal_lang/langspec/LangReader.java:71]: exception?

File: src/main/java/org/mal_lang/langspec/LangReader.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+            readLangSpec(zipIn);
+          } else if (name.startsWith("icons/")) {
+            readIcon(zipIn, name);
+          } else if (name.equals("LICENSE")) {
+            this.license = readString(zipIn);
+          } else if (name.equals("NOTICE")) {
+            this.notice = readString(zipIn);
+          }
+        }
+        zipIn.closeEntry();
```
Comment (mikaelforeseeti) [inline src/main/java/org/mal_lang/langspec/LangReader.java]: Hm, I think java now has some form of open with that gives auto close

Comment (maxwalls) [reply src/main/java/org/mal_lang/langspec/LangReader.java]: zip files are special. `zipIn` is a `ZipInputStream` that is closed at the end of scope on line 109. `closeEntry()` `Closes the current ZIP entry and positions the stream for reading the  next entry`. It seems like `getNextEntry()` that `Reads the next ZIP file entry and positions the stream at the beginning of the entry data` on line 94 calls `closeEntry()`, so I'll remove this.

File: src/main/java/org/mal_lang/langspec/LangReader.java (hunk 1 above)
Comment (maxwalls) [reply src/main/java/org/mal_lang/langspec/LangReader.java:71]: I don't think it should be an error to have other files in the archive. This `return` just ignores files under `icons/` that don't end with `.png` or `.svg`. I just noticed the typo on line 68, I'll fix that.

**Pull request merged**